benchmark/atoi.cpp

```diff
@@ -194,7 +207,14 @@ auto PowersOf10Array() {
     return rv;
 };
 
-template<typename Return>
+
+template<
+    typename Return,
+    std::make_unsigned_t<Return>
+        (*CB10)(
```

Comment:
For a _brief_ moment I was confused at `CB10`. "Is this a star wars robot?" "Callback 10?" no... "Convert to Base 10?"... _ah yes probably..._

I felt a little daft for that not clicking immediately. But, it did get me thinking. 

I think in the context of this specific file, as it is today, this name is probably fine. However, `CB10` could be clearer since it's part of the API of the function.

People may not be viewing the function in the context of the whole file. They may see the functionIntelliSense or whatever language server they are using, and all users will see is `CB10`.

Perhaps this won't be super clear to users of this function. 

Anyway. This got me thinking about the obfuscation as a signifier for (un-)importance. This comment kind of started becoming a whole email about those thoughts, so I'll send that to you in an email. 

Wow, this comment sure did get side-tracked.

Anyway, I'm sure this thought gives us something to discuss tomorrow (today!).

Reply:
This is just the name of a function pointer parameter.  I'm not to blame for the confusing syntax of function pointers.  This is not meant to be user interface either.
I just needed to factor the calculation function out of the body of the conversion function.
There is more work, starting with not assuming the conversion is to base 10.